src/items/type-aliases.md

```diff
@@ -34,9 +34,16 @@ let _ = TypeAlias(5); // Doesn't work
 A type alias without the [_Type_] specification may only appear as an
 [associated type] in a [trait].
 
+A type alias with the [_Type_] specification may only appear as an
+[associated type] in a [trait impl].
```

Comment:
This clause by itself doesn't seem to be correct to me.  It seems to imply that the following is not allowed:

```rust
type X = i32;
```

Perhaps this could be turned around so that it specifically says what a trait impl associated type needs? A sentence could be added to the previous paragraph, since they are talking about the same thing.  Maybe something like this:

A type alias without the [_Type_] specification may only appear as an [associated type] in a [trait]. 
Associated types in a [trait impl] must include the [_Type_] specification.

Reply:
I've made a separate paragraph for each for of type aliases, to be specific about what is allow, what is required, and what isn't allowed in each.